zerver/data_import/hipchat.py

```diff
@@ -0,0 +1,236 @@
+import json
+import logging
+import os
+import subprocess
+
+from typing import Any, Dict, List
+
+from django.conf import settings
+from django.forms.models import model_to_dict
+from django.utils.timezone import now as timezone_now
+
+from zerver.models import (
+    Stream,
+    UserProfile,
+)
+
+from zerver.data_import.import_util import (
+    build_realm,
+    build_recipients,
+    build_zerver_realm,
+)
+
+# stubs
+ZerverFieldsT = Dict[str, Any]
+
+def untar_input_file(tar_file: str) -> str:
+    data_dir = tar_file.replace('.tar', '')
+    data_dir = os.path.abspath(data_dir)
+
+    if os.path.exists(data_dir):
+        logging.info('input data was already untarred to %s, we will use it' % (data_dir,))
+        return data_dir
+
+    os.makedirs(data_dir)
+
+    subprocess.check_call(['tar', '-xf', tar_file, '-C', data_dir])
+
+    logging.info('input data was untarred to %s' % (data_dir,))
+
+    return data_dir
+
+def read_user_data(data_dir: str) -> List[ZerverFieldsT]:
+    fn = 'users.json'
+    data_file = os.path.join(data_dir, fn)
+    data = json.load(open(data_file))
+    return data
+
+def clean_user_data(raw_data: List[ZerverFieldsT]) -> List[ZerverFieldsT]:
+    flat_data = [
+        d['User']
+        for d in raw_data
+    ]
+
+    flat_data = [
+        d
+        for d in flat_data
+        if not d['is_deleted']
+    ]
+
+    def is_realm_admin(v: str) -> bool:
+        if v == 'user':
+            return False
+        elif v == 'admin':
+            return True
+        else:
+            raise Exception('unexpected value')
+
+    '''
+    TODO:
+        avatar
+        timezone
+    '''
+
+    def process(in_dict: ZerverFieldsT) -> ZerverFieldsT:
+        out_dict = dict()
+
+        out_dict['is_realm_admin'] = is_realm_admin(in_dict['account_type'])
+        out_dict['email'] = in_dict['email']
+        out_dict['delivery_email'] = in_dict['email']
+        out_dict['id'] = in_dict['id']
+        out_dict['short_name'] = in_dict['mention_name']
+        out_dict['full_name'] = in_dict['name']
+
+        # unmapped fields:
+        #    title - Developer, Project Manager, etc.
+        #    rooms - no good sample data
+        #    created - we just use "now"
+        #    roles - we just use account_type
+        return out_dict
+
+    return list(map(process, flat_data))
+
+def user_profile_dict(in_dict: ZerverFieldsT, realm_id: int) -> ZerverFieldsT:
+    now = int(timezone_now().timestamp())
+    obj = UserProfile(
+        realm_id=realm_id,
+        date_joined=now,
+        pointer=-1,
+        avatar_source='U',
+        **in_dict
+    )
+    dct = model_to_dict(obj)
+    return dct
```

Comment:
I would probably just integrate this bit into the `process` function above; it seems like the `process` function could do this formatting as well, right?